ballerina/tests/test.bal

```diff
@@ -0,0 +1,243 @@
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testCreateRole() returns error? {
+    ProjectRole response = check jira->/rest/api/'3/role.post(
+        payload = {
+            name: "Developer",
+            description: "Role for developers"
+        }
+    );
+
+    test:assertEquals(response.name, "Developer", "Role name does not match.");
+    test:assertEquals(response.description, "Role for developers", "Role description does not match.");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testDeleteRole() returns error? {
+    int roleId = 10009; 
+    http:Response response = check jira->/rest/api/'3/role/[roleId].delete();
+
+    test:assertEquals(response.statusCode, 204, "Expected 204 No Content status for successful deletion.");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testCreateProject() returns error? {
+    ProjectIdentifiers response = check jira->/rest/api/'3/project.post
+    (
+        payload=
+        {
+            key: "EX",
+            name: "Example",
+            projectTypeKey: "business",
+            leadAccountId:"712020:5d71be5f-debd-474f-9ec9-a97b9023ea4e"
+        }
+    );
+
+    test:assertEquals(response.id, 10001, "Unsuccessful project creation");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testGetProjectSecurityLevelLive() returns error? {
+    string projectKeyOrId = "EX";  
+    ProjectIssueSecurityLevels response = check jira->/rest/api/'3/project/[projectKeyOrId]/securitylevel();
+
+    test:assertEquals(response.levels[0].id, "10000", "No security levels found for project.");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testGetApplicationRoles() returns error? {
+    ApplicationRole[] response = check jira->/rest/api/'3/applicationrole();
+
+    test:assertNotEquals(response.length(), 0, "Expected non-empty application roles array");
+    test:assertEquals(response[0].name, "Jira Software", "Application role name should not be empty");
+    test:assertEquals(response[0].key, "jira-software", "Application role key should not be empty");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testGetSpecificApplicationRole() returns error? {
+    string ProjectKey = "jira-software";
+    ApplicationRole response = check jira->/rest/api/'3/applicationrole/[ProjectKey];
+
+    test:assertEquals(response.key, "jira-software", msg = "Expected key to be 'jira-software'");
+    test:assertEquals(response.name, "Jira Software", msg = "Expected name to be 'Jira Software'");
+}
+
+@test:Config {
+    groups: ["live_tests", "mock_tests"]
+}
+function testGetAuditRecords() returns error? {
+    AuditRecords response = check jira->/rest/api/'3/auditing/'record();
+  
+    test:assertNotEquals(response.records, null, "Expected 'records' array to be non-empty");
```

Comment:
Shall we assert the fields in the returned record?

Reply:
will assert the total number field